openlibrary/plugins/worksearch/schemes/works.py

```diff
@@ -530,21 +530,20 @@ def add_non_solr_fields(self, non_solr_fields: set[str], solr_result: dict) -> N
 
         for doc in solr_result['response']['docs']:
             for ed_doc in doc.get('editions', {}).get('docs', []):
-                ed = ed_key_to_record.get(ed_doc['key'])
-                assert ed
-                for field in non_solr_fields:
-                    val = getattr(ed, field)
-                    if field == 'providers':
-                        provider = get_book_provider(ed)
-                        if not provider:
+                if ed := ed_key_to_record.get(ed_doc['key']):
```

Comment:
Let's add a comment here that `ed` could be `None` if the record has been deleted, but solr hasn't been updated yet. Flipping this to `continue` might reduce some of the nesting, but up to you.